hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -1842,7 +1842,7 @@ private void countDownChildren(RootProcedureState<TEnvironment> procStack,
       store.update(parent);
       scheduler.addFront(parent);
       LOG.info("Finished subprocedure pid={}, resume processing parent {}",
-          procedure.getProcId(), parent);
+          procedure.getProcId(), parent.toString().replace("pid=","ppid="));
```

Comment:
Oh, I see. Good find. How about `"Finished subprocedure (pid={}), resume processing of parent (ppid={})`, and use `parent.getProcId()` instead of the string replace.

Reply:
I think we need other attributes about parent procedure besides pid ?
And replace pid to ppid is a better way than rebuild info string of procedure( rewrite toString method ).

Reply:
Other attributes of the parent should be available via earlier log lines; the operator can search for that `pid=` in the same log.

I still don't like doing a string replace on the output of another class's `toString` method. A format string like `info("... parent ({})", parent);` works as a compromise for me.

Reply:
Yes, I have modified it like this